gatsby-node.esm.js

```diff
@@ -0,0 +1,168 @@
+const path = require('path');
```

Comment:
There are no code differences between here at the previous `gatsby-node.js`. I just moved all of that code into this file for `esm`.

Reply:
In a future PR we can move all the `require`s and `exports` to es syntax 🎉